src/systems/physics/Physics.cc

```diff
@@ -1132,6 +1141,9 @@ void PhysicsPrivate::RemovePhysicsEntities(const EntityComponentManager &_ecm)
       [&](const Entity &_entity, const components::Model *
           /* _model */) -> bool
       {
+        if (this->eventManager)
```

Comment:
Can the GUI not also use `EachRemoved` instead of relying on this event being emitted?